### PR TITLE
Handling Null ref with ScreenId

### PR DESCRIPTION
**Pull request by NavneetThekkumpat**
Resolving Null ref with ScreenId.
Also improving stresstest to include all test .msapps.

Validation: Roundtrips all test .msapps in test battery and in this repo without errors. 
 Roundtrips validation successful for customer .msapp with this inconsistency.

### CODE REVIEW

File: src/PAModelTests/RoundtripTests.cs [hunk 1]
```diff
@@ -13,37 +13,21 @@ namespace PAModelTests
...
-        [DataRow("LocaleBasedApplication.msapp")]
-        public void StressTestApps(string filename)
+        [TestMethod]
+        public void StressTestApps()
         {
-            var root = Path.Combine(Environment.CurrentDirectory, "Apps", filename);
+            var directory = Path.Combine(Environment.CurrentDirectory, "Apps");
+            foreach (var root in Directory.GetFiles(directory))
+            { 
+                Assert.IsTrue(File.Exists(root));
```
Comment (jt000) [inline src/PAModelTests/RoundtripTests.cs:22]: I'd suggest putting the contents of the foreach in a try\catch where the catch is an `Assert.Fail(ex.ToString())` so you can loop through all even when one of the cases throws

File: src/PAModel/SourceTransforms/AppTestTransform.cs [hunk 2]
```diff
@@ -188,7 +188,13 @@ public void BeforeWrite(BlockNode control)
                 // Lookup screenID by Name
                 if (screenProp != null)
                 {
-                    _screenIdToScreenName.ToDictionary(kvp => kvp.Value, kvp => kvp.Key).TryGetValue(screenProp.Expression.Expression, out screenId);
+                    foreach (var prop in _screenIdToScreenName.ToDictionary(kvp => kvp.Value, kvp => kvp.Key))
```
Comment (jt000) [inline src/PAModel/SourceTransforms/AppTestTransform.cs:191]: why "toDictionary" if you're just going to look at _screenIdToScreenName's KVPs? Seems like you could leave that off entirely & just swap the Key for Value below...

File: src/PAModelTests/RoundtripTests.cs [hunk 3]
```diff
@@ -13,37 +13,27 @@ namespace PAModelTests
...
-
-            var cloneOk = MsAppTest.TestClone(root);
-            // If this fails, to debug it, rerun and set a breakpoint in DebugChecksum().
-            Assert.IsTrue(cloneOk, $"Clone failed: " + filename);
+                    var cloneOk = MsAppTest.TestClone(root);
+                    // If this fails, to debug it, rerun and set a breakpoint in DebugChecksum().
+                    Assert.IsTrue(cloneOk, $"Clone failed: " + root);
+                }
+            }
+            catch (Exception ex) {
```
Comment (jt000) [inline src/PAModelTests/RoundtripTests.cs:34]: I'd suggest putting this inside the foreach. This allows for all apps to be tested even when one fails

File: src/PAModelTests/RoundtripTests.cs [hunk 4]
```diff
@@ -13,37 +13,27 @@ namespace PAModelTests
...
+        [TestMethod]
+        public void StressTestApps()
         {
-            var root = Path.Combine(Environment.CurrentDirectory, "Apps", filename);
+            var directory = Path.Combine(Environment.CurrentDirectory, "Apps");
+            try
+            {
+                foreach (var root in Directory.GetFiles(directory))
+                {
+                    Assert.IsTrue(File.Exists(root));
```
Comment (jt000) [inline src/PAModelTests/RoundtripTests.cs:24]: If Directory.GetFiles returned the file, then I don't think this could ever return `false`

**Review by landonarogers — APPROVED** [review]

**Pull request merged**